crates/sui-framework/src/natives/address.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::legacy_create_signer_cost;
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::{account_address::AccountAddress, u256::U256};
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::{
+    loaded_data::runtime_types::Type, natives::function::NativeResult, pop_arg, values::Value,
+};
+use smallvec::smallvec;
+use std::collections::VecDeque;
+
+const E_ADDRESS_PARSE_ERROR: u64 = 0;
+
+const E_CANT_CONVERT_ADDRESS_TO_U256: u64 = 1;
+
+// Implementation of the Move native function address_from_bytes(bytes: vector<u8>): address;
+pub fn from_bytes(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 1);
+
+    let addr_bytes = pop_arg!(args, Vec<u8>);
+
+    // TODO: what should the cost of this be?
+    let cost = legacy_create_signer_cost();
+
+    // Address parsing can fail if fed the incorrect number of bytes.
+    Ok(match AccountAddress::from_bytes(addr_bytes) {
+        Ok(addr) => NativeResult::ok(cost, smallvec![Value::address(addr)]),
+        Err(_) => NativeResult::err(cost, E_ADDRESS_PARSE_ERROR),
+    })
+}
+
+/// Implementation of Move native function `address::to_u256(address): u256`
+pub fn to_u256(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 1);
+
+    let addr = pop_arg!(args, AccountAddress);
+    let mut u256_bytes = [0; 32];
+    let mut addr_bytes_le = addr.to_vec();
+    addr_bytes_le.reverse();
+    // convert the address into a u256 by padding out the lower 12 bytes with 0's
+    for (i, b) in addr_bytes_le.into_iter().enumerate() {
+        u256_bytes[i] = b
+    }
+    let u256_val = Value::u256(U256::from_le_bytes(&u256_bytes));
```

Comment:
Could be simplified a little bit, I think, using `Vec::resize`:

```suggestion
    let mut addr_bytes_le = addr.to_vec();
    addr_bytes_le.reverse();
    addr_bytes_le.resize(32, 0);
    let u256_val = Value::u256(U256::from_le_bytes(&addr_bytes_le));
```